packages/terra-clinical-header/tests/jest/Header.test.jsx

```diff
@@ -1,50 +1,63 @@
 import React from 'react';
 import Header from '../../src/Header';
 
+const consoleSpy = jest.spyOn(global.console, 'warn');
```

Comment:
We should utilize the `beforeAll`, `afterAll` and `afterEach` functions so that the mocks are properly restored and reset after each use.

e.g.:

https://github.com/cerner/terra-framework/blob/5b5d493c6e6b122e18c13e670b85e2abdb1c76f5/packages/terra-worklist-data-grid/tests/jest/WorklistDataGrid.test.jsx#L45-L59

Reply:
[Updated](https://github.com/cerner/terra-clinical/pull/823/commits/a516df0d6cef43cfa1b1b0d6b3978fd479461d1f) : 
with reference to https://jestjs.io/docs/jest-object#jestspyonobject-methodname added `restoreAllMocks()`